### PR TITLE
Jimo Browser Destination

### PR DESCRIPTION
**Pull request by imsamdez**
Jimo enables product teams to create engaging in-app interactions (modal, snippet, product tours, etc). To do so, our customers inject a javascript code into their app.

This PR add a Browser Destination for Jimo. It's responsible for : 
- Injecting the script
- Enrich Jimo user's profile using segment.identify
 
## Testing

We are currently unable to provide test as we are having typescript errors while running test. One of our 3rd party lib, Lottie, make the test crash. Tried different options like disabling all code execution but I'm still having the issue. Seems to happen at build time or something. Everything is working fine in the e2e env.

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/browser-destinations/destinations/jimo/package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  "main": "./dist/cjs",
+  "module": "./dist/esm",
+  "scripts": {
+    "build": "yarn build:esm && yarn build:cjs",
+    "build:cjs": "tsc --module commonjs --outDir ./dist/cjs",
+    "build:esm": "tsc --outDir ./dist/esm"
+  },
+  "typings": "./dist/esm",
+  "dependencies": {
+    "@segment/browser-destination-runtime": "^1.4.0"
```
Comment (zikaari) [inline packages/browser-destinations/destinations/jimo/package.json:19]:
```suggestion
    "@segment/browser-destination-runtime": "^1.14.0"
```

This version needs to pinned to latest runtime version. There's an open PR that'll make sure that every new scaffolded destination will have the latest version to begin with.

https://github.com/segmentio/action-destinations/blob/main/packages/browser-destination-runtime/package.json#L3

File: packages/browser-destinations/destinations/jimo/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    await deps.resolveWhen(
+      () =>
+        Object.prototype.hasOwnProperty.call(window, 'jimo') &&
+        // When initialized on load, window.jimo is first an array then become an object after it get initialized
+        settings.initOnLoad === true &&
+        Array.isArray(window.jimo) === false,
+      100
+    )
+    console.log('Jimo added to window')
```
Comment (zikaari) [inline packages/browser-destinations/destinations/jimo/src/index.ts:56]: Remnants of development aid?

File: packages/browser-destinations/destinations/jimo/src/sendUserData/index.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        '@path': '$.traits.email'
+      }
+    }
+  },
+  defaultSubscription: 'type = "identify"',
+  perform: (jimo, { payload }) => {
+    if (payload.userId != null) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+      jimo.push(['set', 'user:id', [payload.userId]])
+      console.debug('user id sent to jimo')
```
Comment (zikaari) [inline packages/browser-destinations/destinations/jimo/src/sendUserData/index.ts:35]: More remnants

**Review by zikaari — CHANGES_REQUESTED** [review]

**Comment by imsamdez** [comment]
Cleaned up 👍

File: packages/browser-destinations/destinations/jimo/src/sendUserData/index.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import type { Payload } from './generated-types'
+
+const action: BrowserActionDefinition<Settings, JimoSDK, Payload> = {
+  title: 'Send User Data',
+  description: 'Set Jimo identify data from segment.io data',
+  platform: 'web',
+  fields: {
+    userId: {
+      label: 'Internal User Id',
+      description: 'The unique user id provided to segment that will be reuse in Jimo',
```
Comment (joe-ayoub-segment) [inline packages/browser-destinations/destinations/jimo/src/sendUserData/index.ts:13]: Hi @imsamdez can we reword this to make it clearer please?

File: packages/browser-destinations/destinations/jimo/src/sendUserData/index.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      type: 'string',
+      allowNull: true,
```
Comment (joe-ayoub-segment) [inline packages/browser-destinations/destinations/jimo/src/sendUserData/index.ts:15]: Should this field be required or not?
You can set this using the 'required' attribute

File: packages/browser-destinations/destinations/jimo/src/sendUserData/index.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      default: {
+        '@path': '$.userId'
+      }
+    },
+    email: {
+      label: 'User email',
+      description: 'The email of the user',
+      type: 'string',
+      allowNull: true,
```
Comment (joe-ayoub-segment) [inline packages/browser-destinations/destinations/jimo/src/sendUserData/index.ts:24]: Should this field be required or not?
You can set this using the 'required' attribute.

File: packages/browser-destinations/destinations/jimo/src/sendUserData/index.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import type { BrowserActionDefinition } from '@segment/browser-destination-runtime/types'
+import { JimoSDK } from 'src/types'
+import type { Settings } from '../generated-types'
+import type { Payload } from './generated-types'
+
+const action: BrowserActionDefinition<Settings, JimoSDK, Payload> = {
+  title: 'Send User Data',
+  description: 'Set Jimo identify data from segment.io data',
```
Comment (joe-ayoub-segment) [inline packages/browser-destinations/destinations/jimo/src/sendUserData/index.ts:8]: Can this be reworded to explain a little more about what this Action does please?

**Review by joe-ayoub-segment — CHANGES_REQUESTED** [review]
Hi @imsamdez , 

Thanks for raising this PR. 

I've emailed you to request a meeting, which you can schedule via this link: http://calendly.com/joe_ayoub/integration-workshop

Items to discuss: 
1. Let's clarify which events you need sent to Jimo. When we met last you indicated that you wanted user profile data and user analytics events sent to Jimo. The code doesn't really cover these use-cases at the moment. 
2. We need to include unit tests or the PR will not pass CI. When you scaffold a new Integration the tests get create automatically, so you should have a starting point to build on. You can always look at the tests written by other Partners to see how they do it. 
3. Let's also discuss how you are connecting the client library, and if we should be throwing an error if the loadScript function call fails. 
4. We should also add presets and defaultSubscriptions. We can discuss this when we meet. 
5. Field descriptions should be a little more helpful for the customer. 

I look forward to catching up soon,
Kind regards,
Joe

**Comment by imsamdez** [comment]
Tests has been added, let me know if I can help with anything else!

**Review by joe-ayoub-segment — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
hi @imsamdez this PR has been deployed.
